sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierTest.cs

```diff
@@ -26,6 +27,13 @@ public void RawIdTakesPrecendenceInEqualityCheck()
 
             Assert.AreEqual(new PhoneNumberIdentifier("+override", "4:14255550123"), new PhoneNumberIdentifier("14255550123"));
             Assert.AreEqual(new PhoneNumberIdentifier("14255550123"), new PhoneNumberIdentifier("+override", "4:14255550123"));
+
```

Comment:
can we add tests for the following scenarios:

- if the bot id is null or empty
- if the isGlobal is not provided, it should be false
- the default cloud is public one

Reply:
It is impossible to create a MicrosoftBotIdentifier with empty/null id. It will throw an exception.
I added tests for the second two proposals

Reply:
so we have to test if the exception will be thrown for null or empty param, makes it sense?

Reply:
We're not testing it for any other identifier as we rely that the  Argument.AssertNotNullOrEmpty always performs the check.
@petrsvihlik do I need to add additional tests for all identifiers to check that input parameters are validated by  Argument.AssertNotNullOrEmpty?

Reply:
my intention here was not about whether the Argument.AssertNotNullOrEmpty validates or not, it's more about the logic within the constructor, there should be a check for the botId param, and if the botId is invalid it should throw an exception and it will also be a guard for the future from removing the check from constructor.

Reply:
As this is similar to other identifiers (they have the same constructor check), let's wait for the input from @petrsvihlik on whether we need to add and additional test. If we're to add one, ot should be consistent for other identifiers, too.

Reply:
the constructor's fine. @AikoBB just wants to add tests to check that if you provide the constructor with null or empty, it'll throw and that if you initialize it with default values, it'll be set to global+public. I'm fine with adding such tests.

Reply:
Added new tests